agent/metrics/monitor/prometheus_monitor.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package monitor
+
+import (
+	"context"
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/aws/amazon-ecs-agent/agent/dockerclient/dockerapi"
+	"github.com/cihub/seelog"
+	dockercontainer "github.com/docker/docker/api/types/container"
+	"time"
+)
+
+const (
+	PrometheusContainerName = "Prometheus_Monitor"
+	dockerTimeout           = 2 * time.Minute
+)
+
+// Performs the required set-up for the Prometheus container. This function is
+// only called if PrometheusMetrics is enabled in the cfg. We perform the
+// following steps:
+// 1) Create Volume if not created.
+// 2) Create Prometheus container if not created.
+// 3) Start Prometheus container
+func InitPrometheusContainer(ctx context.Context, cfg *config.Config, dg dockerapi.DockerClient) string {
+	if !cfg.PrometheusMetricsEnabled {
+		return ""
+	}
+	prometheusVolumeSetup(dg, cfg, ctx)
+
+	containerID := cfg.PrometheusMonitorContainerID
+
+	// We should create the container in 2 situations: if containerID in state file
+	// was never initialized (Prometheus container never started) or containerID
+	// cannot be inspected. Note the fallthrough keyword here.
+	switch {
+	case containerID != "":
+		_, err := dg.InspectContainer(ctx, containerID, dockerTimeout)
+		if err == nil {
+			seelog.Infof("Prometheus container found with Docker ID: %s", containerID)
+			break
+		}
+		fallthrough
+	case containerID == "":
+		seelog.Info("Prometheus container not found. Creating one.")
+		metadata, err := createPrometheusContainer(dg, cfg, ctx)
+		if err != nil {
+			return ""
```

Comment:
should probably add a log here saying that the create failed and print the err?

Reply:
Added more error logging